plugins/modules/dcnm_links.py

```diff
@@ -0,0 +1,3326 @@
+            None
+        """
+
+        processed_fabrics = []
+
+        if [] is self.config:
+            return
+
+        # Soure fabric is already processed. Add it to processed list
+        processed_fabrics.append(self.fabric)
+
+        for cfg in self.config:
+
+            # For every fabric included in the playbook, get the inventory details. This info is required
+            # to get ip_sn, hn_sn and sn_hn details
+            if cfg.get("dst_fabric", "") != "":
+                if cfg["dst_fabric"] not in processed_fabrics:
+                    processed_fabrics.append(cfg["dst_fabric"])
+                    inv_data = get_fabric_inventory_details(
+                        self.module, cfg["dst_fabric"]
+                    )
+                    self.inventory_data.update(inv_data)
+
+        # Based on the updated inventory_data, update ip_sn, hn_sn and sn_hn objects
+        self.ip_sn, self.hn_sn = get_ip_sn_dict(self.inventory_data)
+        self.sn_hn = dict([(value, key) for key, value in self.hn_sn.items()])
+
+    def dcnm_links_translate_playbook_info(self, config, ip_sn, hn_sn):
+
+        """
+        Routine to translate parameters in playbook if required.
+            - This routine converts the hostname information included in
+              playbook to actual addresses.
+            - translates template names based on version of DCNM
+
+        Parameters:
+            config - The resource which needs tranlation
+            ip_sn - IP address to serial number mappings
+            hn_sn - hostname to serial number mappings
+
+        Returns:
+            None
+        """
+
+        if [] is config:
+            return
+
+        for cfg in config:
+
+            if cfg.get("src_device", "") != "":
+                cfg["src_device"] = dcnm_get_ip_addr_info(
+                    self.module, cfg["src_device"], ip_sn, hn_sn
+                )
+            if cfg.get("dst_device", "") != "":
+                cfg["dst_device"] = dcnm_get_ip_addr_info(
+                    self.module, cfg["dst_device"], ip_sn, hn_sn
+                )
+
+            if cfg.get("template", None) is not None:
+                cfg["template"] = self.templates.get(
+                    cfg["template"], "dcnm_links_invalid_template"
+                )
+
+
+def main():
+
+    """ main entry point for module execution
+    """
+    element_spec = dict(
+        src_fabric=dict(required=True, type="str"),
+        config=dict(required=False, type="list", elements="dict", default=[]),
+        state=dict(
+            type="str",
+            default="merged",
+            choices=["merged", "deleted", "replaced", "query"],
+        ),
+        deploy=dict(type="bool", default="true"),
+    )
+
+    module = AnsibleModule(
+        argument_spec=element_spec, supports_check_mode=True
+    )
+
+    dcnm_links = DcnmLinks(module)
+
+    state = module.params["state"]
+
+    if [] is dcnm_links.config:
+        if state == "merged":
```

Comment:
This should also be mandatory for `replaced` right?

Reply:
yes. you are right. Will update